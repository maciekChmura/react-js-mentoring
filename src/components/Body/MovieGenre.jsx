import React, { Component } from 'react';
import styled from 'styled-components';

const MovieGenreWrapper = styled.p`
  font-size: 10px;
  color: dodgerblue;
  margin: 0px;
`;
const MovieGenre = props => {
  return (
    <MovieGenreWrapper>
      {props.genres.map(genre => {
        return <span key={genre}>{`${genre} `}</span>;
      })}
    </MovieGenreWrapper>
  );
};

export default MovieGenre;
