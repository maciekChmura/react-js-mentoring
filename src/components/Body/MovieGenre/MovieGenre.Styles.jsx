import styled from 'styled-components';

const MovieGenreWrapper = styled.p`
  font-size: 10px;
  color: black;
  margin: 0px;
`;

export default MovieGenreWrapper;
