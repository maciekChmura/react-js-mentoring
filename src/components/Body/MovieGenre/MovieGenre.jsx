import React from 'react';
import MovieGenreWrapper from './MovieGenre.Styles';

const MovieGenre = ({ genres }) => (
  <MovieGenreWrapper>
    {genres.map(genre => (
      <span key={genre}>{`${genre} `}</span>
    ))}
  </MovieGenreWrapper>
);

export default MovieGenre;
