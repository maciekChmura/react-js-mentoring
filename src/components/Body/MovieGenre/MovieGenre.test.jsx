import React from 'react';
import { shallow } from 'enzyme';

import MovieGenre from './MovieGenre';

describe('MovieGenre', () => {
  const genres = ['Animation', 'Family'];

  it('should render correctly', () => {
    const component = shallow(<MovieGenre genres={genres} />);

    expect(component).toMatchSnapshot();
  });
});
