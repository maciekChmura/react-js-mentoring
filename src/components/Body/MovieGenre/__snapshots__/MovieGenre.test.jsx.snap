// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieGenre should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieGenre
    genres={
      Array [
        "Animation",
        "Family",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <span>
          Animation 
        </span>,
        <span>
          Family 
        </span>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "Animation",
        "nodeType": "host",
        "props": Object {
          "children": "Animation ",
        },
        "ref": null,
        "rendered": "Animation ",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "Family",
        "nodeType": "host",
        "props": Object {
          "children": "Family ",
        },
        "ref": null,
        "rendered": "Family ",
        "type": "span",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "MovieGenreStyles__MovieGenreWrapper-sc-8mpqj1-0",
        "isStatic": true,
        "rules": Array [
          "font-size:10px;color:black;margin:0px;",
        ],
      },
      "displayName": "MovieGenreStyles__MovieGenreWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "MovieGenreStyles__MovieGenreWrapper-sc-8mpqj1-0",
      "target": "p",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <span>
            Animation 
          </span>,
          <span>
            Family 
          </span>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "Animation",
          "nodeType": "host",
          "props": Object {
            "children": "Animation ",
          },
          "ref": null,
          "rendered": "Animation ",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "Family",
          "nodeType": "host",
          "props": Object {
            "children": "Family ",
          },
          "ref": null,
          "rendered": "Family ",
          "type": "span",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "MovieGenreStyles__MovieGenreWrapper-sc-8mpqj1-0",
          "isStatic": true,
          "rules": Array [
            "font-size:10px;color:black;margin:0px;",
          ],
        },
        "displayName": "MovieGenreStyles__MovieGenreWrapper",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "MovieGenreStyles__MovieGenreWrapper-sc-8mpqj1-0",
        "target": "p",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
